### PR TITLE
LakeBTC Trading API

### PR DESCRIPTION
**Pull request by cristianlucaci**
Implementation for LakeBTC trading API #687

### CODE REVIEW

**Comment by cristianlucaci** [comment]
Sorry, forgot to upload the jar on nexus. working on it now.

File: xchange-lakebtc/pom.xml [hunk 1]
```diff
@@ -25,6 +25,12 @@
 	<dependencies>
 
 		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-lang3</artifactId>
+			<version>3.3.2</version>
+		</dependency>
```
Comment (timmolter) [inline xchange-lakebtc/pom.xml]: Please put the dependency under `dependencyManagement` in the parent pom.xml.

Comment (cristianlucaci) [inline xchange-lakebtc/pom.xml]: Done, but I have difficulties uploading the jar to nexus, in case you can help with that, I would appreciate it.

Comment (timmolter) [inline xchange-lakebtc/pom.xml]: Did you also remove the version line from `xchange-lakebtc/pom.xml`?

What jars are you trying to upload?  I don;t understand what you mean? Do you mean the Travis CI build is failing and you need help?

Comment (cristianlucaci) [inline xchange-lakebtc/pom.xml]: @timmolter not sure what version from pom I am suppsed to remove, the file was generated before me, I just added commons-io. 

Yes, I was referring to Travi CI build, the examples module needs the new lakebtc jar in order to compile.

Comment (timmolter) [inline xchange-lakebtc/pom.xml]: you need to delete the line:

    <version>3.3.2</version>

in `xchange-lakebtc/pom.xml`

**Comment by timmolter** [comment]
if you run `mvn clean verify` locally does the build also fail?

I suspect the lake-btc module isn't added to `xchange-examples/pom.xml` as a dependency, but that's just a guess.

File: xchange-lakebtc/pom.xml (hunk 1 above)
Comment (cristianlucaci) [inline xchange-lakebtc/pom.xml]: Fixed it. I've used the same dependency as the one from xchange-btcchina\pom.xml, so I suppose that one has the same problem.

**Comment by timmolter** [comment]
Any idea why this cannot be automatically merged? Can yo try pulling and merging to the latest commit on the develop branch?

**Comment by cristianlucaci** [comment]
I've fixed the problems from the CI build fail, but I am not sure why it can not be merged now. Will take a look now.

**Comment by cristianlucaci** [comment]
@timmolter Hi, I tried to merge in develop and it works fine for my, would you need me to create a new pull request from develop instead of this one? I guess the merging problem is due to the last change in pom.xml , where I removed some empty spaces.

**Comment by timmolter** [comment]
@cristianlucaci I think this should work. Are you sure you pulled the latest develop branch and merged it with `cristi/lakebtc`?

**Comment by cristianlucaci** [comment]
@timmolter Yes, I did a pull on develop followed by a merge, and its working fine (a had to manually perform the merge but I didnt got any merge conflict). I can create a new pull request from my fork/develop and close this one, if you agree.

**Comment by sutra** [comment]
Just git push now would work.

**Comment by timmolter** [comment]
Yup, push again and it should work. If not then yeah, another clean PR would be the best. Thanks!

**Comment by cristianlucaci** [comment]
Please delete this PR, I managed to break it once again.

**Comment by timmolter** [comment]
Everything  looks fine now. Should I really delete it?

**Comment by cristianlucaci** [comment]
@timmolter I guess I spoke to soon, indeed it looks fine now. Please fell free to push it, if you believe its working fine. 

Anyway, I still have an implementation done for 796.com, so I guess I learned how to do it right next time (hopefully no issues for my next PR)

**Comment by timmolter** [comment]
Done. Thanks!

**Comment by zholmes1** [comment]
Any reason that the generic implementation throws a ```NotAvailableFromExchangeException``` for ```getTrades```?

**Comment by cristianlucaci** [comment]
@zholmes1 The mdata part was already implemented before me, and I guess I missed this method. If time permits, I can take a look and implement it.

**Pull request merged**